gph/src/ripser.h

```diff
@@ -706,11 +729,19 @@ class ripser
                                         index_t n, OutputIterator out) const
     {
         --n;
-        for (index_t k = dim + 1; k > 0; --k) {
+        for (index_t k = dim + 1; k > 2; --k) {
             n = get_max_vertex(idx, k, n);
             *out++ = n;
             idx -= binomial_coeff(n, k);
         }
+
+        double to_sqrt = 8 * idx + 1;
+        n = static_cast<index_t>(std::floor(1.5 + (0.5 * std::sqrt(to_sqrt)))) - 1;
```

Comment:
This formula is exact; numerics don't seem to be an issue but we should test this more rigorously.